drivers/clock_control/clock_control_nrf.c

```diff
@@ -172,9 +173,94 @@ static void set_on_state(uint32_t *flags)
 	irq_unlock(key);
 }
 
+#ifdef CONFIG_CLOCK_CONTROL_NRF_HFINT_CALIBRATION
+
+static void nrf_errata_30_workaround(void)
+{
+	if (!nrf54l_errata_30()) {
```

Comment:
nit: this would fit better outside `nrf_errata_30_workaround()` (why call the function when the errata does not apply?).